app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee

```diff
@@ -977,24 +977,26 @@ ShortFormApplicationService = (
       )
     )
 
-  # this will only return true if the senior requirement is "everyone",
-  # meaning the primary applicant *must* be a senior
-  Service.applicantDoesNotMeetSeniorRequirements = (member = 'applicant') ->
+  # Return true if the listing is an at least 1 senior building AND oldest member in household is not a senior
+  Service.householdDoesNotMeetAtLeastOneSeniorRequirement = ->
+    requirement = Service.listing.Reserved_Community_Requirement || ''
+    return false unless !!requirement.match(/One household member/g)
+    Service.maxHouseholdAge() < Service.listing.Reserved_community_minimum_age
+
+  # This returns true if the listing is an all senior building AND applicant/app member does not meet age requirement
+  Service.applicantDoesNotmeetAllSeniorBuildingRequirements = (member = 'applicant') ->
     listing = Service.listing
     requirement = listing.Reserved_Community_Requirement || ''
-    reservedType = listing.Reserved_community_type || ''
+    return false unless !!requirement.match(/entire household/i)
+
     if _.isString(member)
       # are we evaluating a form value
       age = Service.memberAgeOnForm(member)
     else
       # or evaluating an appMember object
       age = Service.memberAge(member)
-    !!reservedType.match(/senior/i) && !!requirement.match(/entire household/i) &&
```

Comment:
do we still need this check for senior (vs e.g. veteran or developmental disability?)

Reply:
Actually yes this is a good point, it could be veteran for entire household or at least one - forgot that that was a possibility. Will update.